iina/MainWindowController.swift

```diff
@@ -462,6 +466,29 @@ class MainWindowController: NSWindowController, NSWindowDelegate {
     isInFullScreen = true
   }
   
+  // MARK: - Set TouchBar Time Label
+  
+  func setupTouchBarUI() {
+    guard let duration = playerCore.info.videoDuration else {
+      Utility.fatal("video info not available")
+      return
+    }
+    
+    let pad: CGFloat = 16.0
+    sizingTouchBarTextField.stringValue = duration.stringRepresentation
+    if let widthConstant = sizingTouchBarTextField.cell?.cellSize.width, let posLabel = touchBarCurrentPosLabel {
+      if let posConstraint = touchBarPosLabelWidthLayout {
+        posConstraint.constant = widthConstant + pad
+        posLabel.setNeedsDisplay()
+      } else {
+        let posConstraint = NSLayoutConstraint(item: posLabel, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1.0, constant: widthConstant + pad)
+        posLabel.addConstraint(posConstraint)
+        touchBarPosLabelWidthLayout = posConstraint
+      }
+    }
+
+  }
+  
```

Comment:
Please move this function to the end of `NSTouchBarDelegate` extension, i.e. line 1346.

Also add a system version check in `PlayerCore` (Xcode will show a warning and do this for you).